src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -127,7 +127,7 @@ static void ErrorIfInsertPartitionColumnDoesNotMatchSelect(Query *query,
 														   Oid *
 														   selectPartitionColumnTableId);
 static void AddUninstantiatedEqualityQual(Query *query, Var *targetPartitionColumnVar);
-
+static void ErrorIfRouterSelectQueryNotSupported(Query *queryTree);
```

Comment:
Maybe consider the more specific `ErrorIfQueryHasModifyingCTE` since that's everything this does right now?